### ISSUE TITLE
Make it possible to use owner-like generated templates but for unauthenticated users

### ISSUE THREAD

**Issue by cosimtaiuz** [aws-amplify/amplify-cli#7125]
**Is your feature request related to a problem? Please describe.**
I have found it almost impossible to express that I'd like unauthenticated users which are "authorized" via Cognito's unauthenticated role to be able to create records but at the same time let only the unauthenticated user access and see these records.
So far the only solution I've found is to make such records readable by any unauthenticated user:
```
type FavoritedEvent
  @model(queries: null)
  @auth(
    rules: [
      { allow: public, provider: iam, operations: [read, create, delete, update] }
      { allow: private, provider: iam, operations: [read, create, delete, update] }
    ]
  )
  @key(name: "byUser", fields: ["userId", "eventId"])
  @key(name: "byEvent", fields: ["eventId", "userId"])
{
  id: ID!
  eventId: ID!
  userId: ID!
  event: EventData! @connection(fields: ["eventId"])
  user: User! @connection(fields: ["userId"])
}
```

**Describe the solution you'd like**
I would like to be able to say that a given `type` is `[read, create, delete, update]` only for the unauthenticated user who created it. Mind you, unauthenticated users still have a unique, stable ID associated with them because they receive it via Cognito as an unauthenticated user. (By the way I find extremely confusing that this role is called "IAM public / private" when really it's just Cognito's unauthorized or authorized roles)

**Describe alternatives you've considered**
As above I worked around this by making it readable by anybody who's got at least a cognito ID, but it should be readable only by the user who created it, even if they are unauthenticated.

**Additional context**
Basically I'd like there to be more expressivity and complexity available for unauthenticated users (mind you, *not* public ones) as there is for authenticated ones. I want my users to be able to use my app without having to create an account but still keep track of individual users for the simple purpose of offering some information related to themselves only.
